packages/security-mongo-bundle/src/collections/Sessions.collection.ts

```diff
@@ -73,11 +74,23 @@ export class SessionsCollection<T extends ISession>
       },
     });
   }
+
+  async getConfirmationSessionByUserId(
+    userId: UserId,
+    type: string
+  ): Promise<ISession> {
```

Comment:
Maybe we can do a smart find where we can add the 'data' filter. This would allow us to have more flexibility. 
findSession(token, data) something like that.